improver/snow_fraction.py

```diff
@@ -0,0 +1,167 @@
+from improver.utilities.interpolation import interpolate_missing_data
+
+
+class SnowFraction(PostProcessingPlugin):
+    """
+    Calculates a snow-fraction field from fields of snow and rain (rate or
+    accumulation). Where no precipitation is present, the data are filled in from
+    the nearest precipitating point.
+
+    snow_fraction = snow / (snow + rain)
+    """
+
+    def _get_input_cubes(self, input_cubes):
+        """
+        Separates out the rain and snow cubes from the input list and checks that
+            * No other cubes are present
+            * Cubes represent the same time quantity (instantaneous or accumulation length)
+            * Cubes have compatible units
+            * Cubes have same dimensions
+
+        Args:
+            input_cubes (iris.cube.CubeList):
+                Contains exactly two cubes, one of rain and one of snow. Both either
+                rates or accumulations of the same length and of compatible units.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError:
+                If any of the criteria above are not met.
+
+        """
+        if len(input_cubes) != 2:
+            raise ValueError(
+                f"Expected exactly two input cubes, found {len(input_cubes)}"
+            )
+        rain_name, snow_name = self._get_input_cube_names(input_cubes)
+        self.rain = input_cubes.extract(rain_name).merge_cube()
+        self.snow = input_cubes.extract(snow_name).merge_cube()
+        self.snow.convert_units(self.rain.units)
+        if not spatial_coords_match(self.rain, self.snow):
+            raise ValueError("Rain and Snow cubes are not on the same grid")
+        if not self.rain.coord("time") == self.snow.coord("time"):
+            raise ValueError("Rain and Snow cubes do not have the same time coord")
+
+    @staticmethod
+    def _get_input_cube_names(input_cubes):
+        """
+        Identifies the rain and snow cubes from the presence of "rain" or "snow" in
+        the cube names.
+
+        Args:
+            input_cubes (iris.cube.CubeList):
+                The unsorted rain and snow cubes.
+
+        Returns:
+            tuple:
+                rain_name and snow_name, in that order.
+
+        """
+        cube_names = [cube.name() for cube in input_cubes]
+        try:
+            (rain_name,) = [x for x in cube_names if "rain" in x]
+            (snow_name,) = [x for x in cube_names if "snow" in x]
+        except ValueError:
+            raise ValueError(f"Could not find both rain and snow in {cube_names}")
+        if rain_name == snow_name:
+            raise ValueError(
+                f"Failed to find unique Rain and Snow cubes from {cube_names}"
+            )
+        return rain_name, snow_name
+
+    def _calculate_snow_fraction(self):
+        """
+        Calculates the snow fraction data and interpolates to fill in the missing points
+
+        Returns:
+            iris.cube.Cube:
+                Snow fraction cube
```

Comment:
full stop again just so it's consistent with the other doc strings.

Reply:
Is it worth saying Snow fraction interpolated cube?

Reply:
Added full-stop. I don't think "interpolated" is needed here.